### PR TITLE
BAEL-7282: Getting all results at once in a paged query method

### PR DESCRIPTION
**Pull request by manfred106**

### CODE REVIEW

File: persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/StudentCustomQueryRepository.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.baeldung.paging;
+
+import org.springframework.data.domain.*;
+import org.springframework.data.jpa.repository.*;
+
+public interface StudentCustomQueryRepository extends JpaRepository<Student, String> {
+
+    @Query(value = "SELECT stu FROM Student stu LEFT JOIN FETCH stu.school ",
```
Comment (geniuscodes) [inline persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/StudentCustomQueryRepository.java:8]: can we store these on a file as CONSTANTS

File: persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/School.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.baeldung.paging;
+
+import lombok.*;
```
[the remark this answers is not in the document]
Comment (manfred106) [reply persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/School.java]: I have changed all imports to single import.

File: persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/Student.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package com.baeldung.paging;
+
+import lombok.*;
```
[the remark this answers is not in the document]
Comment (manfred106) [reply persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/Student.java]: I have changed all imports to single import.

File: persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/StudentCustomQueryRepository.java (hunk 1 above)
Comment (manfred106) [reply persistence-modules/spring-boot-persistence-4/src/main/java/com/baeldung/paging/StudentCustomQueryRepository.java:10]: Would you like us to do this?

I have searched our current codebase. This is aligned with our current practice in defining custom queries that put the query string with the attribute value.

**Comment by asif-anwar** [comment]
retest this please

**Pull request merged**